### DISCUSSION TITLE
optionally also log to file

### DISCUSSION THREAD

**Discussion by vesper8** [spatie/ray#430] [General]
On many occasions Ray is crashing after receiving too much content it seems. When I launch it again I have no way to look at all that's been missed while Ray was crashed. I have no way to look at the history at all. It would be nice to have the option to also log all incoming Ray messages to a simple log file in the `ray.php` options file.

Of course I know I can create my own wrapper around Ray to do this but I've taken the habit of just adding a lot of `ray()` call throughout my Laravel project(s) and it would be nice if a more robust log was available while performance continues to be improved to lessen the crashing problem.

**Comment by vesper8** [comment]
@freekmurze any chance this request could be considered? I would really really like to be able to print all incoming ray messages to a log file

**Reply by freekmurze** [reply]
Thanks for your idea. Currently, you are the only one asking for this. To keep the package simple, I'm going to hold off on this one for now.

**Comment by vicenterusso** [comment]
Actually that's a good idea when complex apps needs bigger log on proper text editors, for example
